src/lib/core/trigger.cc

```diff
@@ -74,13 +74,26 @@ trigger_run_reverse(struct rlist *list, void *event)
 	return 0;
 }
 
-void
+int
 trigger_fiber_run(struct rlist *list, void *event, double timeout)
 {
 	struct trigger *trigger, *tmp;
 	unsigned trigger_count = 0;
 	struct region *region = &fiber()->gc;
 	RegionGuard guard(region);
+
+	/* Calculating the total number of triggers. */
+	rlist_foreach_entry(trigger, list, link)
+		trigger_count++;
+
+	struct fiber **fibers = (struct fiber **)
+		region_alloc(region, trigger_count * sizeof(struct fiber *));
```

Comment:
I would advise to use `region_alloc_array()` to keep the memory properly aligned.

Reply:
Sure. Good catch!
<details><summary>diff</summary>

```diff
diff --git a/src/lib/core/trigger.cc b/src/lib/core/trigger.cc
index 3234fab9a..ea0b27b2a 100644
--- a/src/lib/core/trigger.cc
+++ b/src/lib/core/trigger.cc
@@ -86,11 +86,12 @@ trigger_fiber_run(struct rlist *list, void *event, double timeout)
        rlist_foreach_entry(trigger, list, link)
                trigger_count++;
 
+       size_t sz;
        struct fiber **fibers = (struct fiber **)
-               region_alloc(region, trigger_count * sizeof(struct fiber *));
+               region_alloc_array(region, struct fiber *, trigger_count, &sz);
        if (fibers == NULL) {
-               diag_set(OutOfMemory, trigger_count * sizeof(struct fiber),
-                        "region", "for trigger fiber pointers");
+               diag_set(OutOfMemory, sz, "region",
+                        "for trigger fiber pointers");
                return -1;
        }
 
```

</details>